### PR TITLE
avoid installing or using git in container

### PR DESCRIPTION
**Pull request by lv-zheng**
Please review the commits.
Issue #4 can be closed if merged.

### CODE REVIEW

File: genpack.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+# generate `.tar.gz` packages from git repos
+
+# set PACKAGEURL
+set_pack_url() {
+	case "$1" in
+	tunasync)	PACKAGEURL=https://github.com/sjtug/tunasync.git ;;
```
Comment (htfy96) [inline genpack.sh:8]: Consider `wget https://github.com/sjtug/tunasync/archive/master.zip`?

**Comment by Endle** [comment]
感觉可以squash成一个commit再发PR
毕竟这些改动都是耦合的

File: genpack.sh (hunk 1 above)
Comment (lv-zheng) [inline genpack.sh:8]: Use direct package download instead of keeping a local git repo? But why not use wget directly in Dockerfile...

**Comment by lv-zheng** [comment]
@Endle 
I did intend to separate these three commits...
The fourth commit is made after I opened the PR, thus unable to squash :-(

File: genpack.sh (hunk 1 above)
Comment (lv-zheng) [inline genpack.sh:8]: Thus, there are three options
1. Keep using git in genpack script
2. Use wget instead in genpack script
3. Scripts suck, use `RUN wget` in Dockerfile instead
Which one should we use?

**Comment by Endle** [comment]
> I did intend to separate these three commits...
> The fourth commit is made after I opened the PR, thus unable to squash :-(

我觉得你可以把这四个commit都squash成一个
我没记错的话，你是可以覆盖 `lv-zheng:topic_issue_4` 的

**Comment by lv-zheng** [comment]
Squashed

File: genpack.sh (hunk 1 above)
Comment (htfy96) [inline genpack.sh:8]: Could you give some reasons for removing git other than simplifying dependency?

Comment (lv-zheng) [inline genpack.sh:8]: @htfy96 
- Installing git in container consumes image space (~30MB), and is unnecessary.
- Using external `.tar.gz` files makes it possible to switch to older/modified/temporary versions of tunasync/mirror-web more easily

What do you mean by "simplifying dependency"?

**Comment by Endle** [comment]
Cool
看了一下，感觉没必要创建两个bash文件。完全可以合并到一起
话说我又有个idea：如果我们用Makefile来写怎么样？（骆神怎么看）

**Comment by lv-zheng** [comment]
small fix needed: unset PACKAGEURL at the head of set_pack_url().

**Comment by Endle** [comment]
突然想到了这个bash脚本的问题：如果当前目录已经有了这个文件，那应该如何处理？用wget下载下来后，应该如何校验？
所以，是不是换成git checkout?

**Comment by lv-zheng** [comment]
@Endle I have foreseen the dirty workaround to exclude .git directory.

**Comment by Endle** [comment]
@lv-zheng 如果这样，那用 git submodule怎么样？

**Comment by lv-zheng** [comment]
tunasync and mirror-web aren't subproject of mirror-docker. If submodules are used, we have to 钦定 an active developer to keep up with submodule upstream all the time, making commits that only update submodules, which is anti version control. Moreover, submodule make it hard for docker users to switch among different versions of tunasync/web and to debug.

**Comment by Endle** [comment]
为什么不钦定呢？我们又不是要跟上游保持同步
如果tunasync发了新版本，我们肯定要测试后才能用于rebiild image。既然这样，手动update submodule有什么问题呢？

**Comment by lv-zheng** [comment]
OK.
Shall we preserve the `git  archive` and `ADD .tar.gz` approach or use another method?

**Comment by lv-zheng** [comment]
不对啊，如果/sjtug/tunasync发了新版本，那一定是测试好的呀

**Comment by Endle** [comment]
Seems to be fixed.

如果还有问题，那就重开一个 issue 吧

**Pull request closed without merging**